File: src/client/java/set/starlev/starredheltix/util/qol/WelcomeMessage.java
package set.starlev.starredheltix.util.qol;

import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
import net.minecraft.client.MinecraftClient;
import net.minecraft.text.ClickEvent;
import net.minecraft.text.HoverEvent;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;
import set.starlev.starredheltix.client.StarredHeltixClient;

public class WelcomeMessage {
    private static final String[] MODERATORS = {"Starlev", "ZurGames", "MegaChromeX", "nik36c"};
    
    public static void register() {
        ClientPlayConnectionEvents.JOIN.register((handler, sender, client) -> {
            if (StarredHeltixClient.CONFIG.general.firstTimeUser) {
                // Mark as not first time user
                StarredHeltixClient.CONFIG.general.firstTimeUser = false;
                StarredHeltixClient.CONFIG.save();
                
                // Show welcome message after 3 seconds on main thread
                client.execute(() -> {
                    new Thread(() -> {
                        try {
                            Thread.sleep(3000);
                            client.execute(() -> showWelcomeMessage(client));
                        } catch (InterruptedException e) {
                            Thread.currentThread().interrupt();
                        }
                    }).start();
                });
            }
        });
    }
    
    private static void showWelcomeMessage(MinecraftClient client) {
        if (client.player == null) return;
        
        String playerName = client.player.getName().getString();
        boolean isModerator = false;
        
        for (String moderator : MODERATORS) {
            if (moderator.equalsIgnoreCase(playerName)) {
                isModerator = true;
                break;
            }
        }
        
        // Welcome message
        client.player.sendMessage(Text.literal("§6§l§k===§6§l Добро пожаловать в настройку мода StarredHeltix! §6§l§k==="), false);
        client.player.sendMessage(Text.literal("§aСпасибо за установку мода! Вот краткий гайд:"), false);
        client.player.sendMessage(Text.literal(""), false);
        
        // Basic commands
        client.player.sendMessage(Text.literal("§e§lОсновные команды:"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix", "Главный раздел настройки мода через чат"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix toggle", "Включить/выключить мод"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix update", "Проверить обновления мода"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix reloadconfig", "Перезагрузить конфигурацию"), false);
        client.player.sendMessage(createClickableCommand("/вход", "Автоввод пароля (установить: /starredheltix config password <пароль>)"), false);
        client.player.sendMessage(createClickableCommand("/яготовлёвал", "Отправить готовность к подземельям в пати чат"), false);
        client.player.sendMessage(createClickableCommand("/d или /в", "Быстрая команда /dh"), false);
        client.player.sendMessage(Text.literal(""), false);
        
        // Features
        client.player.sendMessage(Text.literal("§b§lФункции мода:"), false);
        client.player.sendMessage(Text.literal("§7• Команды пати: !ping, !fps, !coords, !time, !uptime, !dt, !rng, !boykisser"), false);
        client.player.sendMessage(Text.literal("§7• Блокировка слотов (L для режима блокировки)"), false);
        client.player.sendMessage(Text.literal("§7• Уведомления рыбалки и инвентаря"), false);
        client.player.sendMessage(Text.literal("§7• Таймеры абилок: Киркобулус и Увеличение скорости копания"), false);
        client.player.sendMessage(Text.literal("§7• Таймер Древоточеца с настройкой процента"), false);
        client.player.sendMessage(Text.literal("§7• Решатель Трех незнакомцев"), false);
        client.player.sendMessage(Text.literal("§7• Авто-Спринт и фильтры сообщений"), false);
        client.player.sendMessage(Text.literal("§7• Подсветка эндерменов и волков"), false);
        client.player.sendMessage(Text.literal("§7• Вейпоинты из координат в чате"), false);
        client.player.sendMessage(Text.literal("§7• Блокировка Title для супер-редких сообщений"), false);
        client.player.sendMessage(Text.literal("§7• Напоминания о голосовании"), false);
        client.player.sendMessage(Text.literal("§7• Копирование сообщений (Левый Шифт + ЛКМ)"), false);
        client.player.sendMessage(Text.literal(""), false);
        
        // Configuration commands
        client.player.sendMessage(Text.literal("§d§lНастройки:"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix config password", "Установить пароль для автовхода"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix config readyphrase", "Установить фразу готовности"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix filter add", "Добавить фильтр сообщений"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix filter list", "Показать все фильтры"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix voting toggle", "Включить/выключить напоминания о голосовании"), false);
        client.player.sendMessage(Text.literal(""), false);
        
        // Custom binds
        client.player.sendMessage(Text.literal("§9§lПользовательские бинды:"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix binds create", "Создать новый бинд"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix binds setkey", "Назначить клавишу бинду (ESC для сброса)"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix binds list", "Показать все бинды"), false);
        client.player.sendMessage(createClickableCommand("/starredheltix binds delete", "Удалить бинд"), false);
        client.player.sendMessage(Text.literal(""), false);
        
        if (isModerator) {
            // Moderator commands
            client.player.sendMessage(Text.literal("§c§lМодераторские команды:"), false);
            client.player.sendMessage(Text.literal("§c!sh_mute <игрок> <время> <причина> §7- Забулить игрока"), false);
            client.player.sendMessage(Text.literal("§c!sh_unmute <игрок> §7- Разбулить игрока"), false);
            client.player.sendMessage(Text.literal("§c!sh_kick <игрок> <причина> §7- Выкинуть игрока"), false);
            client.player.sendMessage(createClickableCommand("/sh_check " + playerName, "Проверить игрока на наличие мода"), false);
            client.player.sendMessage(Text.literal("§7Команды работают в пати и ЛС"), false);
            client.player.sendMessage(Text.literal(""), false);
        }
        
        client.player.sendMessage(Text.literal("§6Удачной игры! §7Используйте §e/starredheltix §7для настроек"), false);
        client.player.sendMessage(Text.literal("§6§l==============================="), false);
    }
    
    private static Text createClickableCommand(String command, String description) {
        return Text.literal("§e" + command + " §7- " + description)
            .styled(style -> style
                .withClickEvent(new ClickEvent.SuggestCommand(command))
                .withHoverEvent(new HoverEvent.ShowText(Text.literal("§aНажмите для вставки команды")))
                .withColor(Formatting.YELLOW));
    }
}

File: src/main/kotlin/set/starlev/features/misc/WelcomeMessage.kt
package set.starlev.features.misc

import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents
import net.minecraft.client.Minecraft
import net.minecraft.network.chat.Component
import set.starlev.StarredHeltix

object WelcomeMessage {
    private val mc = Minecraft.getInstance()
    private var hasShown = false

    fun register() {
        ClientPlayConnectionEvents.JOIN.register { handler, sender, client ->
            if (StarredHeltix.feature.misc.welcomeMessage == "1" && !hasShown) {
                sendWelcomeMessage()
                StarredHeltix.feature.misc.welcomeMessage = "0"
                StarredHeltix.configManager.saveConfig("welcome-message-shown")
                hasShown = true
            }
        }
    }

    private fun sendWelcomeMessage() {
        val messages = listOf(
            Component.literal("§6§lДобро пожаловать в StarredHeltix! §7(версия 0.0.9)"),
            Component.literal("§e§lОсновные команды:"),
            Component.literal("§b/sh §eили §b/starredheltix §e- открыть меню настроек"),
            Component.literal("§b/sh update §e- проверить обновления, §b/sh update install §e- установить"),
            Component.literal("§e§lБыстрые команды:"),
            Component.literal("§b/вход §e- быстрый вход с паролем"),
            Component.literal("§b/яготовлёвал §e- отправить сообщение готовности"),
            Component.literal("§b/d §eили §b/в §e- быстрый /dh (голосование)"),
            Component.literal("§e§lУправление биндами:"),
            Component.literal("§b/sh binds create <имя> <команда> §e- создать бинд"),
            Component.literal("§b/sh binds delete <имя> §e- удалить бинд"),
            Component.literal("§b/sh binds setkey <имя> <клавиша> §e- назначить клавишу"),
            Component.literal("§b/sh binds list §e- список биндов"),
            Component.literal("§e§lФильтры сообщений:"),
            Component.literal("§b/sh filter add <сообщение> §e- добавить фильтр"),
            Component.literal("§b/sh filter remove <id> §e- удалить фильтр по ID"),
            Component.literal("§e§lРотация и HUD:"),
            Component.literal("§b/sh rotation §e- показать текущую ротацию"),
            Component.literal("§b/sh rotation <yaw> <pitch> §e- установить ротацию"),
            Component.literal("§b/sh hud editor §e- открыть редактор HUD"),
            Component.literal("§e§lКонфигурация:"),
            Component.literal("§b/sh config password <пароль> §e- установить пароль для входа"),
            Component.literal("§b/sh config readyphrase <фраза> §e- установить фразу готовности"),
            Component.literal("§b/sh reset voting §e- сбросить напоминание о голосовании"),
            Component.literal("§b/sh reset config §e- сбросить всю конфигурацию"),
            Component.literal("§aНаслаждайтесь игрой на сервере Heltix Skyblock!")
        )
        messages.forEach { mc.player?.displayClientMessage(it, false) }
    }
}